### ISSUE TITLE
Excerpt files to .pbf instead of .osm, if possible

### ISSUE THREAD

**Issue by carlhiggs** [healthysustainablecities/global-indicators#191]
Currently, we source .pbf format OpenStreetMap data (a compressed storage format) but export to .osm when extracting the portion corresponding to a buffered study region.   The .osm file can be significantly larger in file size than the source .pbf, even if the geographical area of the pbf is considerably larger; also osm files can be quite large (e.g. 50mb to 500mb).   

These seem to be good reasons for re-outputting excerpts to .pbf, instead of converting to .osm.

osmconvert can output [pbf](https://wiki.openstreetmap.org/wiki/Osmconvert)
osm2pgsql can read [pbf](https://osm2pgsql.org/doc/manual.html#getting-and-preparing-osm-data)

The lines that would require changing are here:

https://github.com/global-healthy-liveable-cities/global-indicators/blob/6d5c2f010c755f91356d948054e8cbf7fb6e9102/process/pre_process/02_create_osm_resources.py#L49

https://github.com/global-healthy-liveable-cities/global-indicators/blob/6d5c2f010c755f91356d948054e8cbf7fb6e9102/process/pre_process/02_create_osm_resources.py#L62

The only reason that I can see for not doing this is if the additional compression/uncompression required added appreciably to processing time at either of those two stages.

But its worth looking into, particularly if we seek to store a small file size example study region in our code repository (#189)

**Comment by carlhiggs** [comment]
Actually, the one place that would require a change of file extension from .osm to .pbf is in project setup here:

https://github.com/global-healthy-liveable-cities/global-indicators/blob/6d5c2f010c755f91356d948054e8cbf7fb6e9102/process/pre_process/_project_setup.py#L174

The change in file name would cascade to other scripts.

**Comment by carlhiggs** [comment]
Incidental to the above, there's a redundant/outdated code line below this that is not used elsewhere and can be deleted:

https://github.com/global-healthy-liveable-cities/global-indicators/blob/6d5c2f010c755f91356d948054e8cbf7fb6e9102/process/pre_process/_project_setup.py#L175-L177

**Comment by carlhiggs** [comment]
So I just did a quick comparison of the two methods for a small test region in Manchester and then for Greater Manchester, just for 10 iterations each of the respective commands across the two output filetypes.  This used the following code launched from the python console from the preprocess directory.  The default study region was changed manually in project setup script, first as the test_sm_manchester region (about 75 sqkm), then as Greater Manchester (1276 sqkm; about 16 times larger).


```
import timeit
import subprocess as sp
from _project_setup import *
locale_poly = f"poly_{db}.poly"
feature = f'PG:"dbname={db} host={db_host} port={db_port} user={db_user} password={db_pwd}" {buffered_urban_study_region}'
command = f'python ogr2poly.py {feature} -f "db"'
sp.call(command, shell=True)
command = f"mv {locale_poly} {locale_dir}/{locale_poly}"
sp.call(command, shell=True)
for osm_region in [
	'manchester_osm_20221015.osm',
	'manchester_osm_20221015.pbf',
]:
    print(f'\n{osm_region}')
    command = f'osmconvert "{osm_data}" -B="{locale_dir}/{locale_poly}" -o="{locale_dir}/{osm_region}"'
    print("- osmconvert")
    timeit.timeit('sp.call(command, shell=True)', number=10, globals = globals())
    command = f"osm2pgsql -U {db_user} -l -d {db} --host {db_host} --port {db_port} {locale_dir}/{osm_region} --hstore --prefix {osm_prefix}"
    print("- osm2pgsql")
    timeit.timeit('sp.call(command, shell=True)', number=10, globals = globals())

```

Here are the results (with some logging output from osm2pgsql removed, as it was verbose; wasn't sure of the flag to hide that for now)
```
test_sm_manchester_osm_20221015.osm
- osmconvert
17.518547102999946
- osm2pgsql
[removed noisy output!]
79.53437023100014

test_sm_manchester_osm_20221015.pbf
- osmconvert
12.472907305999797
- osm2pgsql
[removed noisy output!]
72.97080051800003
```
Those times are for 10 iterations; divide by 10 for the average single run time (pretty quick for this small area, which used Greater Manchester's pbf as an input with a bounding box polygon).

So, in terms of speed, pbf actually wins --- at least for a relatively small study region.  Compared to using .pbf format, it was about 1.4 times as long to excerpt the OSM data, and 1.1 times as long to extract .osm data.  Overall, using the .osm format took about 13% longer.

In addition the file size advantage is quite considerable: the osm file (as currently used) is 101 MB; the pbf file is about 5.1MB; that's a huge difference (.pbf is about 20 times smaller)!

The question is, do these benefits scale as study region becomes larger?  Its possible that compress/uncompression may impact performance -- so we'll check for full manchester too:

So, for Greater Manchester:

```
manchester_osm_20221015.osm
- osmconvert
35.17619475399988
- osm2pgsql
[removed noisy output!]
435.1993458479992

manchester_osm_20221015.pbf
- osmconvert
33.88227260399981
- osm2pgsql
[removed noisy output!]
402.88766323599884
```

Using the larger data excerpt for Greater Manchester, conversion took on average 1.03 times as long using .osm format for excerpt, while excerpting to Postgis took 1.08 times longer using .osm format.

So, some minor speed efficiencies in the favor of .pbf; however for file size: .osm is 616 MB, compared to 27.7MB using .pbf; again, that's (more than) 20 times smaller using .pbf!  So its a bit of a nobrainer to switch to using this format for the excerpts.

There are limitations to these tests --- I was using my computer (including QGIS, multiple browsers open in Chrome, Notepad++ and OneNote) while running them, so there's background noise.  But given there were 10 iterations, it presumably gives at least some chance for differences in individual runs somewhat to be glossed over.
